lib/hdt.js

```diff
@@ -64,31 +89,31 @@ HdtDocumentPrototype.search = HdtDocumentPrototype.searchTriples;
 
 module.exports = {
   // Creates an HDT document for the given file.
-  fromFile: function (filename, callback, self) {
-    if (typeof callback !== 'function') return;
+  fromFile: function (filename) {
     if (typeof filename !== 'string' || filename.length === 0)
-      return callback.call(self, Error('Invalid filename: ' + filename));
+      return Promise.reject(Error('Invalid filename: ' + filename));
 
-    // Construct the native HdtDocument
-    hdtNative.createHdtDocument(filename, function (error, document) {
-      // Abort the creation if any error occurred
-      if (error) {
-        switch (error.message) {
-        case 'Error opening HDT file for mapping.':
-          return callback.call(self, Error('Could not open HDT file "' + filename + '"'));
-        case 'Non-HDT Section':
-          return callback.call(self, Error('The file "' + filename + '" is not a valid HDT file'));
-        default:
-          return callback.call(self, error);
+    return new Promise((resolve, reject) => {
+      hdtNative.createHdtDocument(filename, function (error, document) {
+        // Abort the creation if any error occurred
+        if (error) {
+          switch (error.message) {
+          case 'Error opening HDT file for mapping.':
+            return reject(new Error('Could not open HDT file "' + filename + '"'));
+          case 'Non-HDT Section':
+            return reject(new Error('The file "' + filename + '" is not a valid HDT file'));
+          default:
+            reject(error);
```

Comment:
`return` missing?
And if so, how did the unit tests miss this?

Reply:
A `reject` handler from a Promise constructor does not need to be returned, it only needs to be called. Added the return here for consistency

Reply:
It's not about returning a value, but rather about terminating the function right there. Otherwise, execution continues, and that's not what you want. It's not observable in this particular case though, but we reject because we don't want execution to continue.